tests/test_update_ssh_keys.rs

```diff
@@ -0,0 +1,705 @@
+            true,         // is_replace
+        ) {
+            Ok(keys) => add_key_check_results(ssh_dir, keys, pubkeyvalid2, testnametwo, assert_keys),
+            Err(err) => panic!("update_ssh_keys --add failed {}", err),
+        }
+
+        // "update-ssh-keys --disable two"
+        let sshkeyname1 = "valid1";
+        let sshkeyname2 = "valid2";
+
+        let mut aks = open_authorized_keys(ctx.ssh_dir.to_str().unwrap());
+        let keyfiles = [format!("{}/{}.pub", ctx.ssh_dir.to_str().unwrap(), sshkeyname2)];
+
+        let mut keys = vec![];
+        for keyfile in keyfiles {
+            let file = File::open(&keyfile).expect(format!("failed to open keyfile '{:?}'", keyfile).as_str());
+            keys.append(&mut AuthorizedKeys::read_keys(file).unwrap_or_default());
+        }
+
+        for key in aks.disable_keys("two") {
+            if let AuthorizedKeyEntry::Valid {
+                ref key,
+            } = key
+            {
+                assert!(key.to_fingerprint_string().contains(&*FINGERPRINTS[sshkeyname2]));
+
+                aks.write().expect("failed to update authorized keys directory");
+                aks.sync().expect("failed to update authorized keys");
+
+                let authkeyone: PathBuf = PathBuf::from(format!("{}/authorized_keys.d/{}", ctx.ssh_dir.to_str().unwrap(), "two"));
+                assert!(authkeyone.exists());
+
+                let mut svec: Vec<&str> = Vec::new();
+                svec.push(sshkeyname1);
+                assert_has_keys(ctx.ssh_dir.to_str().unwrap(), svec, true);
+            }
+        }
+
+        // add two again
+        // "update-ssh-keys --add two valid2.pub"
+        //
+        // NOTE: Ideally adding two again here is supposed to immediately fail,
+        // but for some reason it simply hangs forever. For now do not run the
+        // following command to proceed to the next tests.
+
+        //        let assert_keys = ["valid1"].to_vec();
+        //        match add_one_ssh_key(
+        //            ssh_dir,      // ssh_dir
+        //            pubkeyvalid2, // pubkeyname
+        //            testnametwo,  // testname
+        //            false,        // is_force
+        //            true,         // is_replace
+        //        ) {
+        //            Ok(keys) => add_key_check_results(ssh_dir, keys, pubkeyvalid2, testnametwo, assert_keys),
+        //            Err(err) => panic!("update_ssh_keys --add failed {}", err),
+        //        }
+    }
+
+    #[test]
+    fn test_disable() {
+        let ctx = setup_tests();
+
+        run_test_disable(ctx);
+    }
+
+    #[test]
+    fn test_enable() {
+        let ctx = setup_tests();
+
+        let ssh_dir = ctx.ssh_dir.to_str().unwrap();
+        let pubkeyvalid2 = "valid2";
+        let testnametwo = "two";
+        let assert_keys = ["valid1", "valid2"].to_vec();
+
+        // "update-ssh-keys --disable two"
+        run_test_disable(ctx.clone());
```

Comment:
```suggestion
        run_test_disable(&ctx);
```
Pass by reference, the cloning causes the removal race condition that fails the test

Reply:
Changed the type of the function parameter to `&PathBuf`.